File: docs/guides/java/src/test/java/com/algolia/searchWithLogicalOr.java
package com.algolia;

import com.algolia.api.SearchClient;
import com.algolia.config.*;
import com.algolia.model.search.*;
import java.util.List;

public class searchWithLogicalOr {

  public static void main(String[] args) throws Exception {
    try (SearchClient client = new SearchClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")) {
      String query = "the query";

      OptionalWords optionalWords = OptionalWords.of(List.of("the", "query"));

      SearchParams searchParams = new SearchParamsObject().setQuery(query).setOptionalWords(optionalWords);

      client.searchSingleIndex("<YOUR_INDEX_NAME>", searchParams, Hit.class);
    } catch (Exception e) {
      System.out.println("An error occurred: " + e.getMessage());
    }
  }
}

File: docs/guides/kotlin/src/main/kotlin/com/algolia/snippets/searchWithLogicalOr.kt
import com.algolia.client.api.SearchClient
import com.algolia.client.configuration.*
import com.algolia.client.extensions.*
import com.algolia.client.model.search.*
import com.algolia.client.transport.*

suspend fun searchWithLogicalOr() {
  val client = SearchClient(appId = "ALGOLIA_APPLICATION_ID", apiKey = "ALGOLIA_API_KEY")
  val query = "the query"
  val optionalWords = listOf("the", "query")
  val searchParams =
    SearchParamsObject(query = query, optionalWords = OptionalWords.of(optionalWords))
  client.searchSingleIndex(indexName = "<YOUR_INDEX_NAME>", searchParams = searchParams)
}
